java/core/src/java/org/apache/orc/impl/ParserUtils.java

```diff
@@ -244,80 +250,172 @@ public static TypeDescription parseType(ParserUtils.StringPosition source) {
 
   private static final Pattern INTEGER_PATTERN = Pattern.compile("^[0-9]+$");
 
+  public static ColumnVector[] findBranchVectors(TypeDescription schema,
```

Comment:
I think this would be better named findColumnVectors.

Reply:
agreed